### ISSUE TITLE
Support list for Zookeeper URL for Solr cloud and index job

### ISSUE THREAD

**Issue by pluradj** [JanusGraph/janusgraph#1165]
Reported on [janusgraph-users](https://groups.google.com/d/msg/janusgraph-users/x5CnYVpr1ls/BNfwNQ9dCAAJ)

```
15:44:17 WARN  org.apache.hadoop.hbase.zookeeper.RecoverableZooKeeper  - Unable to create ZooKeeper Connection
java.net.UnknownHostException: [app01: invalid IPv6 address
```

Where the graph configuration properties

```
index.search.solr.zookeeper-url=app01:2181,app02:2181,app03:2181
```

The [`ZOOKEEPER_URL`](https://github.com/JanusGraph/janusgraph/blob/0.2/janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/SolrIndex.java#L161-L163) is defined as a `String` property rather than a `String[]` (for example, `storage.hostname` and `index.[X].solr.http-urls` use `String[]`). If the property allows a list, it would allow for a quorum of Zookeeper nodes to be specified.

**Comment by jerryjch** [comment]
The error shown is related to HBase, but the config shown is for Solr?
Both HBase and Solr Zookeeper URL use a string, but both have the string as concatenation of the multiple host and/or port in the same string.  JanusGraph passes the string to HBase or Solr client, not a list, and not parsing the string.
This is working.

**Comment by genganpeng** [comment]
Wheh execute MapReduceIndexManagement，it is error！ other thing is ok。
maybe bug below code of MapReduceIndexManagement：
        String k = ConfigElement.getPath(JanusGraphHadoopConfiguration.GRAPH_CONFIG_KEYS, true) + "." + key;
       String v = source.getProperty(key).toString();//zookeeper address is list, toString() become [], then throw error

**Comment by hoda1369** [comment]
I have the same issue when I execute MapReduceIndexManagement I get the same error. I am using Hbase as backend and Solr as my indexing I have 3 zookeeper.quorum. I can not only specify one zookeeper it throws different error. Any suggestions?

**Comment by pluradj** [comment]
@hoda1369 what different error does it throw? please provide details.

**Comment by hoda1369** [comment]
I have to change storage.hostname to only one host  to not get the error when I execute MapReduceIndexManagement. But I get different error which is:
Exception in thread "main" java.lang.NoClassDefFoundError: com/sun/jersey/api/client/config/ClientConfig
	at org.apache.hadoop.yarn.client.api.TimelineClient.createTimelineClient(TimelineClient.java:59)
	at org.apache.hadoop.yarn.client.api.impl.YarnClientImpl.createTimelineClient(YarnClientImpl.java:185)
	at org.apache.hadoop.yarn.client.api.impl.YarnClientImpl.getTimelineDelegationToken(YarnClientImpl.java:353)
	at org.apache.hadoop.yarn.client.api.impl.YarnClientImpl.addTimelineDelegationToken(YarnClientImpl.java:330)
	at org.apache.hadoop.yarn.client.api.impl.YarnClientImpl.submitApplication(YarnClientImpl.java:250)
	at org.apache.hadoop.mapred.ResourceMgrDelegate.submitApplication(ResourceMgrDelegate.java:291)
	at org.apache.hadoop.mapred.YARNRunner.submitJob(YARNRunner.java:291)
	at org.apache.hadoop.mapreduce.JobSubmitter.submitJobInternal(JobSubmitter.java:240)
	at org.apache.hadoop.mapreduce.Job$10.run(Job.java:1290)
	at org.apache.hadoop.mapreduce.Job$10.run(Job.java:1287)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:422)
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1866)
	at org.apache.hadoop.mapreduce.Job.submit(Job.java:1287)
	at org.apache.hadoop.mapreduce.Job.waitForCompletion(Job.java:1308)
	at org.janusgraph.hadoop.scan.HadoopScanRunner.runJob(HadoopScanRunner.java:137)
	at org.janusgraph.hadoop.MapReduceIndexManagement.updateIndex(MapReduceIndexManagement.java:187)
	at com.att.hypergraph.core.GraphManager.reIndex(GraphManager.java:266)
	at com.att.hypergraph.core.TestGraphManagmentDriver.main(TestGraphManagmentDriver.java:41)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.SparkSubmit$.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:750)
	at org.apache.spark.deploy.SparkSubmit$.doRunMain$1(SparkSubmit.scala:187)
	at org.apache.spark.deploy.SparkSubmit$.submit(SparkSubmit.scala:212)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:126)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
Caused by: java.lang.ClassNotFoundException: com.sun.jersey.api.client.config.ClientConfig
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:338)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 28 more

**Comment by genganpeng** [comment]
this maybe jar version problem, note NoClassDefFoundError

**Comment by jerryjch** [comment]
[This](https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!topic/janusgraph-users/iJVk2EU08HM) is a related and new thread on the mailing list with more and clear details.

**Comment by jerryjch** [comment]
Fixed by PR #1282
